src/libswami/SwamiPlugin.c

```diff
@@ -105,7 +105,7 @@ _swami_plugin_initialize(void)
     }
     else
     {
-        plugin_paths = g_list_append(plugin_paths, PLUGINS_DIR);
+        plugin_paths = g_list_append(plugin_paths, g_strdup(PLUGINS_DIR));
```

Comment:
`g_strdup` may return NULL. It might be better to bind `PLUGINS_DIR` to a `static const char*`, then add that constant to the `plugin_paths` and only call `g_free()` if `data != static const char*` .

Reply:
Ok.